collection-pipeline/src/test/java/com/iluwatar/collectionpipeline/AppTest.java

```diff
@@ -29,27 +29,75 @@
 import java.util.List;
 import java.util.Map;
 
-import org.junit.jupiter.api.Test;
+import org.junit.Assert;
+import org.junit.jupiter.api.*;
 
 /**
  * Tests that Collection Pipeline methods work as expected.
  */
 public class AppTest {
-  
+
   private List<Car> cars = CarFactory.createCars();
-  
+
+  @BeforeAll
+  public static void beforeAll() {
+    int branches = 12;
+    Car.visitedBranchGlobal = new boolean[branches];
+    Car.visitedBranchLocal = new boolean[branches];
+  }
+
+  @BeforeEach
+  public void beforeEach() {
+    Arrays.fill(Car.visitedBranchLocal, false);
+  }
+
+  /**
+   * Runs after each test. Adds newly covered branches to global coverage.
+   */
+  @AfterEach
+  public void afterEach() {
+    boolean verbose = false;
+
+    for (int i = 0; i < Car.visitedBranchLocal.length; i++) {
+      if (Car.visitedBranchLocal[i]) {
+        Car.visitedBranchGlobal[i] = true;
+      }
+    }
+    if (verbose) {
+      for (int i = 0; i < Car.visitedBranchLocal.length; i++) {
+        System.out.format("Branch %d covered: %b.\n", i, Car.visitedBranchLocal[i]);
+      }
+    }
+  }
+
+  /**
+   * Prints global test coverage for whole test suite.
+   */
+  @AfterAll
+  public static void afterAll() {
+    int cover = 0;
+    for (int i = 0; i < Car.visitedBranchGlobal.length; i++) {
+      System.out.format("Branch %d covered: %b.\n", i + 1, Car.visitedBranchGlobal[i]);
+      if (Car.visitedBranchGlobal[i]) {
+        cover++;
+      }
+    }
+    System.out.println("Test done, results:");
+    System.out.format("Branch cover is: %f.\n", (double) cover / Car.visitedBranchGlobal.length);
```

Comment:
The percentage will never be 100%, cover must either be cover + 1 och start at 1 at the beginning of afterAll().